IETF.js

```diff
@@ -0,0 +1,355 @@
+{
+	"translatorID": "ac4b1d6a-6e52-42e2-9634-ebbe1e96b9ad",
+	"label": "IETF",
+	"creator": "Félix Brezo (@febrezo)",
+	"target": "^https?://tools.ietf.org/(html|pdf|rfc)/",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2020-11-09 00:18:41"
+}
+
+/*
+	IETF Translator
+	Copyright (C) 2020 Félix Brezo (@febrezo), felixbrezo@disroot.org
+	
+	This program is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+	
+	You should have received a copy of the Affero GNU General Public License
+	along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+function detectWeb(doc, url) {
+	return "report";
+}
+
+function doWeb(doc, url) {
+	var resourceType = detectWeb(doc, url);
+	var newItem = new Zotero.Item(resourceType);
+	
+	var targetUri;
+	if (url.includes("datatracker")) {
+		let auxTitle = ZU.xpathText(doc, "//title");
+		newItem.institution = "IETF";
+		newItem.title = auxTitle.split(" - ")[1];
+		newItem.reportNumber = auxTitle.split(" - ")[0];
+		
+		let allContent = ZU.xpathText(doc, "//pre");
+		try {
+			let re = /Category: (.+)  +[^\n]+?/;
+			newItem.reportType = allContent.match(re)[1];
+		} catch (catError) {
+			let re = /Category: (.+)\n/;
+			newItem.reportType = allContent.match(re)[1];
+		}
+		
+		let auxAuthors = ZU.xpathText(doc, "((//table[@class='table table-condensed']/tbody/tr)[2]/td)[2]").split(',');
```

Comment:
This isn't a good XPath to use. There's no need for the extra classes or the `tbody`, which make this unnecessarily fragile, and just going by row index is dangerous — e.g., they could easily change the 'table-condensed' class or add an extra row. In this case, it would probably be better to use something like `//table//th[text()="Authors"]/following-sibling::td/following-sibling::td`, unless the 'Authors' label is localized. Still fragile — it's dependent on a hidden `td` with class `edit`, though that could be ignored — but a bit less so.

Scraping still isn't great, though. Why not just retrieve the BibTeX for these and import it via the BibTeX translator? Even if some data had to be adjusted after ("RFC Editor" → "IETF"), that would provide a more stable and complete base. Grep`9cb7002` in the repo root for examples.

The 8557 example is missing the date and abstract, so that has to be fixed one way or another. These should also all set a DOI, as shown in the BibTeX. (That won't currently show up in Scaffold but will be saved to the Extra field for later migration to a real DOI field for Report.)

(It looks like old RFCs have a warning about incomplete BibTeX, so scraping could potentially be used for those — not sure if there's a specific numeric cutoff or we'd have to just look for the warning comment.)

Reply:
Thanks for the comments @dstillman. I have just added the date and the abstract for the new example which was missing. Anyway, the usage of the Bibtex would have simplified many of the selections made.  I'll try to find an example of another translator that deals with Bibtex links found inside the website for an easier import.

I have also considered your recommendations about the usage of XPath Axes like `following-sibling` and so on. In any case, I still needed to apply a regular expression to grab the abstract in the Datatracker since the abstract text is out of any XML node and cannot specifically grab it with decendants, siblings, parents and so on. I still feel that the Bibtex approach may help here a lot since, in fact, the abstract seems to be there.

In any case, I think that this needs another update to include the Bibtex. I'll do my best to find a good example and I'll try to add it for a more consistent approach probably during the weekend.